plugin/evm/customheader/gas_limit.go

```diff
@@ -101,8 +107,22 @@ func VerifyGasLimit(
 	header *types.Header,
 ) error {
 	switch {
+	case config.IsGranite(header.Time):
+		timeMS := *customtypes.GetHeaderExtra(header).TimeMilliseconds
+		state, err := feeStateBeforeBlock(config, parent, timeMS)
+		if err != nil {
+			return fmt.Errorf("calculating initial fee state: %w", err)
+		}
+		maxCapacity := state.MaxCapacity()
+		if header.GasLimit != uint64(maxCapacity) {
+			return fmt.Errorf("%w: have %d, want %d",
+				errInvalidGasLimit,
+				header.GasLimit,
+				maxCapacity,
+			)
+		}
 	case config.IsFortuna(header.Time):
-		state, err := feeStateBeforeBlock(config, parent, header.Time)
+		state, err := feeStateBeforeBlock(config, parent, header.Time*1000)
```

Comment:
Check https://github.com/ava-labs/coreth/pull/1288#issuecomment-3346446701

```suggestion
    case config.IsFortuna(header.Time):
        state, err := feeStateBeforeBlock(config, parent, customtypes.HeaderTimeMilliseconds(header))
```